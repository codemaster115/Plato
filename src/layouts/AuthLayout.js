import React from 'react';
import PropTypes from 'prop-types';
import Paper from '@material-ui/core/Paper';
import CssBaseline from '@material-ui/core/CssBaseline';
import styled from 'styled-components';

export const Main = styled.main`
    display: block;
    width: 400px;
    margin: auto;
`;

export const StyledPaper = styled(Paper)`
    margin-top: 64px;
    display: flex;
    flex-direction: column;
    align-items: center;
    padding: 16px 0px 24px;
`;

class AuthLayout extends React.Component {
  render() {
    return (
      <Main>
       <CssBaseline />
         <StyledPaper>        
          {this.props.children}
         </StyledPaper>         
      </Main>
    );
  }
}

export default AuthLayout;
