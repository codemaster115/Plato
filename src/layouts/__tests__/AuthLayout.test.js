import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AuthLayout from '../AuthLayout';

describe('AuthLayout', () => {
  it('Compare snapshots', () => {
    const Component = renderer.create(<AuthLayout />);

    const Tree = Component.toJSON();
    expect(Tree).toMatchSnapshot();
  });

  it('Renders without crashing', () => {
    mount(<AuthLayout />);
  });
});
