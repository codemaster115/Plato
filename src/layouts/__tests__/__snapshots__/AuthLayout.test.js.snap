// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout Compare snapshots 1`] = `
.c0 {
  display: block;
  width: 400px;
  margin: auto;
}

.c1 {
  margin-top: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 0px 24px;
}

<main
  className="c0"
>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 c1"
  />
</main>
`;
